ivy/functional/frontends/torch/reduction_ops.py

```diff
@@ -180,6 +180,22 @@ def min(*input, dim=None, keepdim=False, out=None):
         )
 
 
+@numpy_to_torch_style_args
+@to_ivy_arrays_and_back
+def mode(input, dim=-1, keepdim=False, *, out=None):
+    if dim is None:
+        input = ivy.reshape(input, [-1])
+        dim = 0
+    unique_values, counts = ivy.unique_counts(input)
```

Comment:
The unique_count function flattens the entire array. However, to align with the desired functionality, it is essential to calculate the mode for each 1D array of the input array. You would need to find a way to achieve that functionality.